### DISCUSSION TITLE
Link the definition of multiple variables together??

### DISCUSSION THREAD

**Discussion by rtaskf** [adieyal/sd-dynamic-prompts#626] [Q&A]
I want to set a group of variables together, so they have a matching theme.

I tried it like this:

{${var1=warm}${var2=summer}|${var1=cold}${var2=winter}}

... but as soon as i start putting variable definitions inside a {...|...}, compilation doesn't succeed anymore.

What can I do? I need to randomize and then define multiple variables in a branched manner.

**Comment by mddle** [comment]
Not tested, but use @ for cyclical combinations.

Try for text wildcards:
```
 ${var1={warm|cold}} ${var2={summer|winter}}

{@${var2}} is {@${var1}}

summer is warm
winter is cold
```
or for YAML:
```
temp:
  - warm
  - cold
season:
  - summer
  - winter

${var1=__temp__} ${var2=__season__}

{@${var2} is {@$(var1}}

summer is warm
winter is cold

```

When using multiple variables with YAML wildcards, I find it's much easier to use the simple variable.

i.e. don't use ```$(var=!__wildcard__} or $(var=@__wildcard__}```, instead use the default ```${var=__wildcard__}``` then ```{@${var}}``` for cyclical, and ```{~${var}}``` for immediate evaluation.

**Reply by rtaskf** [reply]
Thanks for replying. Yes, I began using cycling and batches a lot. It's not random, but at least it's synced. Particularly useful when you need to sync positive and negative prompts for different styles and such. Immediate evaluation (!=) only seems to make sense when you have to sync the result of that variable for use in other variables, since you can't use them as freely as the normal ones (=).

**Reply by mddle** [reply]
I use this for one random variation of two or more variables:
```
{~{${var1} ${var2}}} ${var3} 
```
So, if var3 is {red|green|blue}, you'll get a random pairing of var1 and var2 for each of var3.

This is much quicker than:
```
{~${var1}} {~${var2}} ${var3}
```
and also works with weights.
